### PR TITLE
Solve player registration conflicts

### PR DESCRIPTION
**Pull request by zbrachinara**

### CODE REVIEW

File: squire_lib/src/players/player_registry.rs [hunk 1]
```diff
@@ -96,21 +102,26 @@ impl PlayerRegistry {
...
-                        let plyr = Player::from_account(account);
-                        let digest = Ok(plyr.id);
-                        self.name_and_id.insert(name, plyr.id);
-                        self.players.insert(plyr.id, plyr);
-                        digest
-                    }
-                }
+                let Some(name) = (!self.name_and_id.contains_left(&account.user_name))
+                    .then_some(account.get_user_name())
+                    .or(tourn_name)
```
Comment (TylerBloom) [inline squire_lib/src/players/player_registry.rs:107]: If we get to this point, we should also check that the `tourn_name` is also not already in use.

File: squire_lib/src/operations/mod.rs [hunk 2]
```diff
@@ -20,8 +20,9 @@ pub use player_ops::PlayerOp;
 #[derive(Serialize, Deserialize, Debug, Hash, Clone, PartialEq, Eq)]
 /// This enum captures all ways in which a tournament can mutate.
 pub enum TournOp {
-    /// Operation for a player register themself for a tournament
-    RegisterPlayer(SquireAccount),
+    /// Operation for a player register themself for a tournament, using a tournament-specific name
+    /// if their user name is already taken
+    RegisterPlayer(SquireAccount, Option<String>),
```
Comment (TylerBloom) [inline squire_lib/src/operations/mod.rs:25]: You'll probably run into a number of issues with this change, and it's not really needed. After a player is registered, the caller can use the returned ID to get the player's name in the tournament.

Comment (zbrachinara) [reply squire_lib/src/operations/mod.rs:25]: To be sure, is this meant for `OpData`?

Comment (TylerBloom) [reply squire_lib/src/operations/mod.rs:25]: Oh ya, my bad. This was meant for `OpData`.

**Comment by TylerBloom** [comment]
Overall, this looks good to me, but there is a missing change in `squire_sdk/src/sync/manager.rs`. You need to add `None`. This should be caught by the SDK CI checks, but there is a missing argument to `cargo hack test`. I'll correct that, but the test(s) need to be corrected. After that, this PR looks good to me.

**Review by TylerBloom — DISMISSED** [review]

**Review by TylerBloom — DISMISSED** [review]

**Review by TylerBloom — APPROVED** [review]

**Pull request merged**